runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -63,8 +69,22 @@ protected ManagementClient getOrchestrator(Layout layout) {
             throw new WorkflowException("getOrchestrator: no available orchestrators " + layout);
         }
 
-        return runtime.getLayoutView().getRuntimeLayout(layout)
+        // Select an available orchestrator
+        ManagementClient managementClient = runtime.getLayoutView().getRuntimeLayout(layout)
                 .getManagementClient(activeLayoutServers.get(0));
```

Comment:
Why do we select an orchestrator in advance? if it will be picked further in the loop..

Reply:
just a "default" orchestrator instead of null;